### DISCUSSION TITLE
Organizing Migrations

### DISCUSSION THREAD

**Discussion by scal-lider** [adonisjs/core#2490] [Help] [answered]
Hi there. 
I have too many migrations T_T, so I would like to organize it in folders.
It is possible? 
As far as I know, the migration runs in [alphabetical order](https://github.com/adonisjs/core/discussions/2345#discussioncomment-449063), if it is possibles, the folder needs to be in alphabetical order too? Or only the files inside folders?

In the example below, wich migration runs first?
```
a/
|-- 1_populate_table
b/
|-- 0_create_table
```

**Comment by e-e** [comment, marked as answer]
This is pretty easy to test - just try it out:

With a folder structure like:

```
database/
|-- migrations/
|---- 00/
|------ 1618549676073_test_3_s
|------ 1618549678710_test_4_s
|---- 01/
|------ 1618549669282_test_1_s
|------ 1618549672676_test_2_s
```

it ran in the following order:
```
+ node ace migration:run
❯ migrated database/migrations/00/1618549676073_test_3_s
❯ migrated database/migrations/00/1618549678710_test_4_s
❯ migrated database/migrations/01/1618549669282_test_1_s
❯ migrated database/migrations/01/1618549672676_test_2_s
```

so yes, it will run all migrations in folder "A" before folder "B"

**Comment by RomainLanz** [comment]
Hey @scal-lider! 👋 

We are using a deep natural sort.

In other terms, your folders and files are run in your expected order. 👍

**Comment by scal-lider** [comment]
So I have to search "natural sort" 😅 
Basically it transforms strings in tuples with strings and number and compare them, so:
```
"test_10" > "test_2"  is true

"test_10" => ["test_", 10]
"test_2" => ["test_", 2]

"test_" == "test_" AND 10 > 2
```

When compare a string with a number, it uses a string comparation:
 `'a' > '1618549676073'`

Based on the @e-e's test, the sort also count the folder's name:
`00/1618549676073_test_3_s  < 01/1618549669282_test_1_s`, 
And **probably**
`a/0_test > 1618549676073_test > 00/1618549676073_test_3_s` 
since `'a' > '1' and 'a' >'0'` and `1618549676073 > 0`

**Comment by Xqua** [comment]
Hey! I just built a tool that might help with this: **[adonis-lucid-migration-squash](https://github.com/[your-username]/adonis-lucid-migration-squash)**

Instead of organizing migrations into folders, you can **consolidate/squash** them into a single baseline migration (similar to Django's `squashmigrations`).

**What it does:**
- Takes your production database schema
- Generates a clean, single Knex migration file
- Includes automated verification to ensure 100% accuracy
- Reduces deployment time (1 migration instead of 100+)

**Quick example:**
```bash
# Dump your schema
pg_dump -s --no-owner --no-acl > schema.sql

# Convert to Knex migration
python -m pg_to_knex schema.sql baseline_migration.ts

# Done! Clean baseline migration with verification
```
